sdk/python/feast/infra/infra_object.py

```diff
@@ -49,15 +57,38 @@ def from_infra_object_proto(infra_object_proto: InfraObjectProto) -> Any:
             infra_object_proto: A protobuf representation of an InfraObject.
 
         Raises:
-            ValueError: The type of InfraObject could not be identified.
+            FeastInvalidInfraObjectType: The type of InfraObject could not be identified.
         """
         if infra_object_proto.infra_object_class_type:
             cls = _get_infra_object_class_from_type(
                 infra_object_proto.infra_object_class_type
             )
             return cls.from_infra_object_proto(infra_object_proto)
 
-        raise ValueError("Could not identify the type of the InfraObject.")
+        raise FeastInvalidInfraObjectType()
+
+    @staticmethod
+    def from_proto(infra_object_proto: Any) -> Any:
+        """
+        Converts a protobuf representation of a subclass to an object of that subclass.
+
+        Args:
+            infra_object_proto: A protobuf representation of an InfraObject.
+
+        Raises:
+            FeastInvalidInfraObjectType: The type of InfraObject could not be identified.
+        """
+        if isinstance(infra_object_proto, DatastoreTableProto):
```

Comment:
I see. Is this what you meant by not needing to change the interface of new online stores?

This seems ok for now. If the user uses feast plan and there's some plugin though, might be worth still outputting the rest of feast plan's output (i.e. FCOs) without throwing an error?

Reply:
Thoughts on using multiple inheritance for this? e.g. https://docs.python.org/3/tutorial/classes.html#multiple-inheritance

Reply:
that's fair - I think the exception should still be thrown here, and the higher level logic that executes feast plan and logs can just not act on this exception if it arises